File: Week_01/maxArea.php
<?php
class Solution {

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height) {
        $max = 0;
        $left = 0;
        $right = count($height) - 1;
        while ($left < $right){
            $area = ($height[$left] > $height[$right] ? $height[$right] : $height[$left]) * ($right - $left);
            if($max < $area){
                $max = $area;
            }
            if($height[$left] >= $height[$right]){
                $right --;
            }else{
                $left ++;
            }
        }
        return $max;
    }
}
$obj = new Solution();
echo $obj->maxArea([1,8,6,2,5,4,8,3,7]);

File: Week_01/maxArea.go
package main

import "fmt"

func maxArea(height []int) int {
	var max, area, left, right = 0, 0, 0, len(height)-1
	for ; left < right ;{
		if height[left] > height[right]{
			area = height[right] * (right - left)
			right--
		} else{
			area = height[left] * (right - left)
			left++
		}
		if area > max{
			max = area
		}
	}
	return max
}

func main(){
	fmt.Println(maxArea([]int{1,8,6,2,5,4,8,3,7}))
}
